spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1051,21 +1051,29 @@ The Reactive Messaging implementation reads the global application configuration
 
 * `mp.messaging.incoming.[channel-name].[attribute]=[value]`
 * `mp.messaging.outgoing.[channel-name].[attribute]=[value]`
+* `mp.messaging.connector.[connector-name].[attribute]=[value]`
```

Comment:
It's the associated messaging provider fully qualified name. I agree that connector-name is better here but it needs to be indicated somewhere.

Reply:
I was thinking that the string used for the [connector-name] should match the value give for the `type` attribute. I agree that 'connector' is a much cleaner string to use in the literal [incoming|outgoing|connector] but I am less comfortable with that literal string being used to indicate the fqClassname of a MessagingProvider.  I think the root of the cause is that have both "Connector" and "MessagingProvider" as synonyms that mean the same thing but you have to 'learn' which is which is not a great design. 
We have:
1 - a marker interface for a Class used as the 'type' which is a key to a set of attributes.
2 - an injectable IncomingConnectorFactory
3 - an injectable OutgoingConnectorFactory
If we are using CDI qualifiers etc to resolve the connector, I wonder about using the
connecter bean to obtain the 'type'. This can be done in a more typesafe manner and
could mean we get rid of the MessagingProvider as a seperate entity and are left with
the simpler notion of a single concept - a connector class.

Reply:
I note that we have got
```
Class<? extends MessagingProvider> provider() default MessagingProvider.class
```
as an attribute in both `@Outgoing` and '@Incoming' annotationsso we can have code like

@Outgoing( "orders_channel", provider="com.acme.Kafka")

I think it is a wise design choice if we allow all parameters of the `@Incoming/@Outgoing` annotation to be overridden with properties that begin: 
```
mp.messaging.incoming.[channel-name].[parameter-name] = [override to annotation value]
```
With no 'special' cases for type/provider 
In which case. 
As we currently have the 'provider' attribute of both the `@Incoming` and `@Outgoing`
annotations we need to allow propeties of: 
```
mp.messaging.outgoing.[channel-name].provider=[qualifier-value-or-fq-classname]
mp.messaging.incoming.[channel-name].provider=[qualifier-value-or-fq-classname]
```
If we add more parameters to the annotations these will be overridable by the same pattern.
Having a different design is less 'orthoganol' and requires user to learn 'special cases.'

Reply:
I would like to rename the `type()` method in `IncomingConnectionFactory` and `OutgoingConnectionFactory` to `provider()` or `getProvider()` I will see if I can
comment on that PR...

Reply:
Basically I have to see how #22 nets out and then come back and edit these comments on top of our use of Qualifiers or however else #22 nets out (as it will remove the provider annotation parameter!